From 0bca0d4d2bf95a37e4a403457ee5c7db4c0b6290 Mon Sep 17 00:00:00 2001
From: Pete Zaitcev <zaitcev@kotori.zaitcev.us>
Date: Wed, 18 Oct 2017 18:15:11 -0500
Subject: [PATCH] Be more tolerant of exception messages from sqlite

Parsing the text of error messages was always perilous, perhaps.
Either way, this should fix the problem of test_get() failing.

Change-Id: I929c2f6e60624241075a292d020c5707ea0ff1c3
---
 swift/common/db.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/swift/common/db.py b/swift/common/db.py
index 91cb919767..ce4def9d02 100644
--- a/swift/common/db.py
+++ b/swift/common/db.py
@@ -333,7 +333,9 @@ def possibly_quarantine(self, exc_type, exc_value, exc_traceback):
             exc_hint = 'malformed'
         elif 'malformed database schema' in str(exc_value):
             exc_hint = 'malformed'
-        elif 'file is encrypted or is not a database' in str(exc_value):
+        elif ' is not a database' in str(exc_value):
+            # older versions said 'file is not a database'
+            # now 'file is encrypted or is not a database'
             exc_hint = 'corrupted'
         elif 'disk I/O error' in str(exc_value):
             exc_hint = 'disk error while accessing'
